packages/gatsby/src/utils/worker/__tests__/pool.ts

```diff
@@ -0,0 +1,231 @@
+  matchPath: undefined,
+  componentChunkName: `component--laia`,
+  isCreatedByStatefulCreatePages: true,
+  context: {
+    var: true,
+  },
+  updatedAt: 1,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  pluginCreator___NODE: `laia`,
+  pluginCreatorId: `laia`,
+  ownerNodeId: `laia`,
+}
+
+const dummyStaticQuery = {
+  id: `sq--book`,
+  name: `book-name`,
+  componentPath: `/static-query-component.js`,
+  query: `{ nodeTypeOne { resolverField } }`,
+  hash: `book-hash`,
+}
+
+const queryIds: IGroupedQueryIds = {
+  pageQueryIds: [dummyPageElias, dummyPageLaia],
+  staticQueryIds: [dummyStaticQuery.id],
+}
+
+describeWhenLMDB(`worker (pool)`, () => {
+  beforeAll(async () => {
+    store.dispatch({ type: `DELETE_CACHE` })
+    const fileDir = path.join(process.cwd(), `.cache/redux`)
+    await fs.emptyDir(fileDir)
+
+    worker = createTestWorker()
+
+    const siteDirectory = path.join(__dirname, `fixtures`, `sample-site`)
+    await loadConfigAndPlugins({ siteDirectory })
+    await worker.loadConfigAndPlugins({ siteDirectory })
+    await sourceNodesAndRemoveStaleNodes({ webhookBody: {} })
+    await getDataStore().ready()
+
+    await build({ parentSpan: {} })
+
+    store.dispatch({
+      type: `CREATE_PAGE`,
+      plugin: {
+        id: `gatsby-plugin-test`,
+        name: `gatsby-plugin-test`,
+        version: `1.0.0`,
+      },
+      payload: {
+        path: dummyPageElias.path,
+        componentPath: dummyPageElias.componentPath,
+        component: dummyPageElias.component,
+      },
+    })
+
+    store.dispatch({
+      type: `CREATE_PAGE`,
+      plugin: {
+        id: `gatsby-plugin-test`,
+        name: `gatsby-plugin-test`,
+        version: `1.0.0`,
+      },
+      payload: {
+        path: dummyPageLaia.path,
+        componentPath: dummyPageLaia.componentPath,
+        component: dummyPageLaia.component,
+      },
+    })
+
+    saveStateForWorkers([`inferenceMetadata`])
+
+    store.dispatch({
+      type: `QUERY_EXTRACTED`,
+      plugin: {
+        id: `gatsby-plugin-test`,
+        name: `gatsby-plugin-test`,
+        version: `1.0.0`,
+      },
+      payload: {
+        componentPath: dummyPageElias.componentPath,
+        query: dummyPageElias.query,
+      },
+    })
+
+    store.dispatch({
+      type: `QUERY_EXTRACTED`,
+      plugin: {
+        id: `gatsby-plugin-test`,
+        name: `gatsby-plugin-test`,
+        version: `1.0.0`,
+      },
+      payload: {
+        componentPath: dummyPageLaia.componentPath,
+        query: dummyPageLaia.query,
+      },
+    })
+
+    store.dispatch({
+      type: `REPLACE_STATIC_QUERY`,
+      plugin: {
+        id: `gatsby-plugin-test`,
+        name: `gatsby-plugin-test`,
+        version: `1.0.0`,
+      },
+      payload: dummyStaticQuery,
+    })
+
+    saveStateForWorkers([`components`, `staticQueryComponents`])
+
+    await worker.buildSchema()
+    await runQueriesInWorkersQueue(worker, queryIds)
```

Comment:
Is this line the only difference between this test and `queries.ts` tests (where `worker.runQueries` is used as opposed to this new function that create batches too)? 

If so I do think we could maybe adjust assertions here to verify that `worker.runQueries` was called N times (and test could probably create over >50 pages to check if batching behaves correctly. And `queries.ts` itself does assert that queries are run correctly?